usb.go

```diff
@@ -210,6 +210,36 @@ func (c *Context) OpenDevices(opener func(desc *DeviceDesc) bool) ([]*Device, er
 	return ret, reterr
 }
 
+func (c *Context) OpenDeviceWithFileDescriptor(fd uintptr) (*Device, error) {
```

Comment:
wouldn't it be better to open a device from a system path? Open() can happen inside the function (note that Open will need to become a method on the libusb interface, so that it can be faked).

Unless you know of an example where this wouldn't be sufficient?